src/Microsoft.IdentityModel.Tokens/Validation/SigningKeyValidationResult.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+#nullable enable
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// Contains the result of validating the <see cref="SecurityKey"/> used to sign a <see cref="SecurityToken"/>.
+    /// The <see cref="TokenValidationResult"/> contains a collection of <see cref="ValidationResult"/> for each step in the token validation.
+    /// </summary>
+    internal class SigningKeyValidationResult : ValidationResult
+    {
+        private Exception? _exception;
+
+        /// <summary>
+        /// Creates an instance of <see cref="SigningKeyValidationResult"/>
+        /// </summary>
+        /// <paramref name="signingKey"/> is the security key that was validated successfully.
+        public SigningKeyValidationResult(SecurityKey? signingKey)
+            : base(ValidationFailureType.ValidationSucceeded)
+        {
+            SigningKey = signingKey;
```

Comment:
Can this actually be null in the success case?

Reply:
There is a case where the TVP have `ValidateIssuerSigningKey` set to `false` and we return in a successful manner while logging a warning about the fact that no validation has been performed.
This is a common pattern across our validators that we will need to decide if we want to differentiate as `NotEvaluated` or something similar. Right now the existing public methods return in the exact same manner if either the validation succeeds or no validation is performed (by not throwing an exception).

Reply:
Also in the case where `TokenValidationParameters` have `RequireSignedTokens` set to `false`, a similar logic follows if the key is null.

Reply:
thanks